File: twoSum/twoSum.cpp
//Sort first then use two pointer techniques


#include <iostream>
#include <vector>
#include <algorithm>



using namespace std;

vector<int> twoSum(vector<int>& nums, int target) 
    {
        cout << "function call " << endl;
    }




int main()
{

    vector<int> usin = {15,7,2,11};
    int target = 9;
    
    sort(usin.begin(),usin.end());

    for (auto i = usin.begin(); i != usin.end(); ++i)
    {
        cout << *i << " ";
    }

    cout << endl;

    int leftPointer = usin[0];
    int rightPointer = usin[usin.size()-1];
    cout << leftPointer << " " << rightPointer << endl;
    int sum = leftPointer + rightPointer;

    vector<int>::iterator it;

    

    return 0;

    // while (sum!=target)
    // {
    //     if (sum > target)
    //     {
    //         return 0;
    //     }
    //     else if (sum < target)
    //     {
    //         return 0;
    //     }
    // }

}

File: twoSum/twoSum.py


def twoSum(nums,target):
    ans = []
    ans += nums
   
    ans.sort()
    
   
    leftPtr = 0
    rightPtr = len(ans)-1
    res = None
    while (leftPtr != rightPtr ):
  
        res = ans[leftPtr] + ans[rightPtr]
        if res > target:
            rightPtr -= 1
        elif res < target:
            leftPtr += 1
        else:
            x = ans[leftPtr]
            y = ans[rightPtr]
            
            a = nums.index(x)
            b = nums.index(y)
            if a == b:
                nums[a] = "buffer"
                b = nums.index(y)
                return [a,b]
            else:
                return [a,b]

        
    return None
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
# print(twoSum([2,7,11,15],9))
# print(twoSum([3,2,4],6))
# print(twoSum([3,3],6))
print(twoSum([3,2,3],6))

